### ISSUE TITLE
Division by large int returns 0 before CPython does

### ISSUE THREAD

**Issue by verhovsky** [RustPython/RustPython#3818]
CPython 3.10

```pycon
>>> 1 / 10**309
1e-309
>>> 1 / 10**310
1e-310
[etc.]
>>> 1 / 10**322
1e-322
>>> 1 / 10**323
1e-323
>>> 1 / 10**324
0.0
```

RustPython [`1b7b3f067c`](https://github.com/RustPython/RustPython/tree/1b7b3f067cd49be55a4788d8e1ad74622e6e7b73)

```none
>>>>> 1 / 10**307
1.0000000000000001e-307
>>>>> 1 / 10**308
1e-308
>>>>> 1 / 10**309
0.0
>>>>> 1e-323
1e-323
```

This is the reason this test is failing: https://github.com/RustPython/RustPython/blob/b8f276f8d76a3c02ccaca19dc638bc545abb5cf7/Lib/test/test_statistics.py#L1477-L1479

**Comment by maong0927** [comment]
Can you assign this issue to me?

**Comment by adam-tokarski** [comment]
Hi @maong0927, do you still work on that issue?

**Comment by bonjune** [comment]
Currently RustPython works as below:
```
>>>>> 1 / 10**309
1e-309
>>>>> 1 / 10**310
1e-310
>>>>> 1 / 10**311
1e-311
>>>>> 1 / 10**320
1e-320
>>>>> 1 / 10**323
1e-323
>>>>> 1 / 10**324
0.0
```

**Comment by DimitrisJim** [comment]
Indeed, seems to working as expected now!

**Comment by verhovsky** [comment]
This was fixed by #4241

**Comment by youknowone** [comment]
Thank you for checking!
